### PR TITLE
Add followVector2 method to Camera

### PR DESCRIPTION
**Pull request by spydon**
# Description

With this method it will be less code duplication when fixing the viewport for `flame_forge2d`.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors and are passing (See [Contributor Guide]).
- [x] I updated/added relevant documentation (doc comments with `///`) and updated/added examples in `doc/examples`.
- [x] I have formatted my code with `./scripts/format.sh` and the Flame analyzer (`./scripts/analyze.sh`) does not report any problems.
- [x] I read and followed the [Flutter Style Guide].
- [x] I have added a description of the change under `[next]` in `CHANGELOG.md`.
- [x] I removed the `Draft` status, by clicking on the `Ready for review` button in this PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flame users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo

### CODE REVIEW

File: packages/flame/lib/src/game/camera.dart [hunk 1]
```diff
@@ -205,7 +205,27 @@ class Camera {
...
+  /// Immediately snaps the camera to start following [vector2].
+  ///
+  /// This means that the camera will move so that the position vector is in a
+  /// fixed position on the screen.
+  /// That position is determined by a fraction of screen size defined by
+  /// [relativeOffset] (default to the center).
+  /// [worldBounds] can be optionally set to add boundaries to how far the
+  /// camera is allowed to move.
+  void followVector2(
+    Vector2 vector2, {
```
Comment (erickzanardo) [inline packages/flame/lib/src/game/camera.dart:224]:
```suggestion
    this.follow, {
```

File: packages/flame/lib/src/game/camera.dart [hunk 2]
```diff
@@ -205,7 +205,27 @@ class Camera {
...
+    Vector2? relativeOffset,
+    Rect? worldBounds,
+  }) {
+    follow = vector2;
```
Comment (erickzanardo) [inline packages/flame/lib/src/game/camera.dart:228]:
```suggestion
```

**Review by erickzanardo — APPROVED** [review]
Two small suggestions but LGTM already

File: packages/flame/lib/src/game/camera.dart (hunk 1 above)
Comment (spydon) [reply packages/flame/lib/src/game/camera.dart:224]: That is only possible in constructors?

Comment (erickzanardo) [reply packages/flame/lib/src/game/camera.dart:224]: Lol, indeed

**Review by luanpotter — APPROVED** [review]
LGTM

**Pull request merged**